In [4]:
import pandas as pd
import numpy as np


In [81]:
url = "https://archive.ics.uci.edu/ml/machine-learning-databases/adult/adult.data"
columns = ['age', 'workclass', 'fnlwgt', 'education', 'education-num',
           'marital-status', 'occupation', 'relationship', 'race', 'sex',
           'capital-gain', 'capital-loss', 'hours-per-week', 'native-country', 'income']

df = pd.read_csv(url, names=columns, na_values='?', skipinitialspace=True)

print("Первые 5 строк данных:")
df.head()

Первые 5 строк данных:


age         workclass  fnlwgt  education  education-num  \
0   39         State-gov   77516  Bachelors             13   
1   50  Self-emp-not-inc   83311  Bachelors             13   
2   38           Private  215646    HS-grad              9   
3   53           Private  234721       11th              7   
4   28           Private  338409  Bachelors             13   

       marital-status         occupation   relationship   race     sex  \
0       Never-married       Adm-clerical  Not-in-family  White    Male   
1  Married-civ-spouse    Exec-managerial        Husband  White    Male   
2            Divorced  Handlers-cleaners  Not-in-family  White    Male   
3  Married-civ-spouse  Handlers-cleaners        Husband  Black    Male   
4  Married-civ-spouse     Prof-specialty           Wife  Black  Female   

   capital-gain  capital-loss  hours-per-week native-country income  
0          2174             0              40  United-States  <=50K  
1             0             0              13  United-States  <=50K  
2             0             0              40  United-States  <=50K  
3             0             0              40  United-States  <=50K  
4             0             0              40           Cuba  <=50K

In [82]:
print(f"1) Число столбцов: {num_columns}")

1) Число столбцов: 15


In [83]:
missing_columns = df.columns[df.isnull().any()].tolist()
print(f"2) Есть ли пропуски в данных? {'Да' if missing_columns else 'Нет'}")
#проверка пропущенных значений
print("\nПропущенные значения в каждом столбце:")
print(df.isnull().sum())


2) Есть ли пропуски в данных? Да

Пропущенные значения в каждом столбце:
age                  0
workclass         1836
fnlwgt               0
education            0
education-num        0
marital-status       0
occupation        1843
relationship         0
race                 0
sex                  0
capital-gain         0
capital-loss         0
hours-per-week       0
native-country     583
income               0
dtype: int64


In [84]:
unique_race_count = df['race'].nunique()
print(f"\n3) Количество уникальных значений в столбце race: {unique_race_count}")


3) Количество уникальных значений в столбце race: 5


In [85]:
median_hours = df['hours-per-week'].median()
print(f"\n4) Медиана hours-per-week: {median_hours}")


4) Медиана hours-per-week: 40.0


In [86]:
high_income = df[df['income'] == '>50K']
gender_count = high_income['sex'].value_counts()
more_gender = gender_count.idxmax()
count_more = gender_count.max()
count_less = gender_count.min()
print(f"\n5) Больше {more_gender} с ЗП >50K")


5) Больше Male с ЗП >50K


In [87]:
print("6) Заполнение пропущенных данных:")
for column in df.columns:
    most_common_value = df[column].mode()[0]
    df[column] = df[column].fillna(most_common_value)
#проверка
missing_data_after = df.isnull().sum()
if missing_data_after.sum() == 0:
    print("   Все пропущенные данные успешно заполнены.")
else:
    print(f"  Остались пропуски:\n{missing_data_after[missing_data_after > 0]}")

6) Заполнение пропущенных данных:
   Все пропущенные данные успешно заполнены.


1) Число столбцов: 15
2) Пропуски в данных: есть

*   workclass         1836
*   occupation        1843
*   native-country     583


3) Кол-во уникальных значений в столбце race: 5
4) Медиана hours-per-week: 40.0
5) Кого больше с ЗП >50K: Male
6) Другие методы заполнения пропусков:

*  Средним значением
*  Медианой
*  Метод интерполяции
* Удаление строк или столбцов с пропусками
* Алгоритмы машинного обучения
